core/src/mindustry/game/SpawnGroup.java

```diff
@@ -157,20 +157,9 @@ public String toString(){
         '}';
     }
 
-    public SpawnGroup copy(){
-        SpawnGroup newGroup = new SpawnGroup();
-        newGroup.type = this.type;
-        newGroup.end = this.end;
-        newGroup.begin = this.begin;
-        newGroup.spacing = this.spacing;
-        newGroup.max = this.max;
-        newGroup.unitScaling = this.unitScaling;
-        newGroup.unitAmount = this.unitAmount;
-        newGroup.shields = this.shields;
-        newGroup.shieldScaling = this.shieldScaling;
-        newGroup.effect = this.effect;
-        newGroup.items = this.items;
-        return newGroup;
+    @Override
+    public SpawnGroup clone() throws CloneNotSupportedException{
```

Comment:
I meant `copy()` should call `clone()` and handle the exception, see how `Weapon` does it.